core/sql/src/main/java/org/trafodion/sql/HDFSClient.java

```diff
@@ -0,0 +1,319 @@
+         else
+         {
+            buf_.limit(bufOffset_ + length_);
+            bytesRead = fsdis_.read(buf_);
+         }
+         return new Integer(bytesRead);
+      }
+   }
+       
+   public HDFSClient() 
+   {
+   }
+ 
+   public HDFSClient(int bufNo, String filename, ByteBuffer buffer, long position, int length) throws IOException
+   {
+      bufNo_ = bufNo; 
+      filename_ = filename;
+      Path filepath = new Path(filename_);
+      fs_ = FileSystem.get(filepath.toUri(),config_);
+      fsdis_ = fs_.open(filepath);
+      blockSize_ = (int)fs_.getDefaultBlockSize(filepath);
+      buf_  = buffer;
+      bufOffset_ = 0;
+      pos_ = position;
+      len_ = length;
+      if (buffer.hasArray()) 
+         bufLen_ = buffer.array().length;
+      else
+      {
+         bufLen_ = buffer.capacity();
+         buf_.position(0);
+      }
+      lenRemain_ = (len_ > bufLen_) ? bufLen_ : len_;
+      if (lenRemain_ != 0)
+      {
+         int readLength = (lenRemain_ > blockSize_) ? blockSize_ : lenRemain_;
+         future_ = executorService_.submit(new HDFSRead(readLength));
+      }
+   }
+
+   public int trafHdfsRead() throws IOException, InterruptedException, ExecutionException
+   {
+      Integer retObject = 0;
+      int bytesRead;
+      int readLength;
+       
+      if (lenRemain_ == 0)
+         return 0;
+      retObject = (Integer)future_.get();
+      bytesRead = retObject.intValue();
+      if (bytesRead == -1)
+         return -1;
+      bufOffset_ += bytesRead;
+      pos_ += bytesRead;
+      lenRemain_ -= bytesRead;
+      if (bufOffset_ == bufLen_)
+         return bytesRead; 
+      else if (bufOffset_ > bufLen_)
+         throw new IOException("Internal Error in trafHdfsRead ");
+      if (lenRemain_ == 0)
+         return bytesRead; 
+      readLength = (lenRemain_ > blockSize_) ? blockSize_ : lenRemain_;
+      future_ = executorService_.submit(new HDFSRead(readLength));
+      return bytesRead;
+   } 
+
+   public int trafHdfsReadBuffer() throws IOException, InterruptedException, ExecutionException
+   {
+      int bytesRead;
+      int totalBytesRead = 0;
+      while (true) {
+         bytesRead = trafHdfsRead();
+         if (bytesRead == -1 || bytesRead == 0)
+            return totalBytesRead;
+         totalBytesRead += bytesRead;
+         if (totalBytesRead == bufLen_)
+              return totalBytesRead;
+      }  
+   } 
+
+   boolean hdfsCreate(String fname , boolean compress) throws IOException
+   {
+     if (logger_.isDebugEnabled()) 
+        logger_.debug("HDFSClient.hdfsCreate() - started" );
+      Path filePath = null;
+      if (!compress || (compress && fname.endsWith(".gz")))
+        filePath = new Path(fname);
+      else
+        filePath = new Path(fname + ".gz");
+        
+      FileSystem fs = FileSystem.get(filePath.toUri(),config_);
+      FSDataOutputStream fsOut = fs.create(filePath, true);
+      
+      if (compress) {
```

Comment:
Since adding a new compression type takes only a few lines of code, it will be good to look into increasing the types of compressed unload files we can write to. I can file a JIRA if there is interest. It is unrelated to focus of this change.